app/src/main/java/de/danoeh/antennapod/activity/SplashActivity.java

```diff
@@ -33,7 +35,9 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
             DrawableCompat.setTint(wrapDrawable, 0xffffffff);
             progressBar.setIndeterminateDrawable(DrawableCompat.unwrap(wrapDrawable));
         } else {
-            progressBar.getIndeterminateDrawable().setColorFilter(0xffffffff, PorterDuff.Mode.SRC_IN);
+            final ColorFilter colorFilter = BlendModeColorFilterCompat
+                    .createBlendModeColorFilterCompat(0xffffffff, BlendModeCompat.SRC_IN);
+            progressBar.getIndeterminateDrawable().setColorFilter(colorFilter);
```

Comment:
Why do you need a BlendMode color filter (with compatibility)? Before it was PorterDuff, which can be added without the compat library: `new PorterDuffColorFilter(0xffffffff, Mode.SRC_IN)`

Reply:
The BlendMode compatibility simply allows us to use the newer colorfilter method on OS versions that support it: see [here](https://developer.android.com/reference/androidx/core/graphics/BlendModeColorFilterCompat#createBlendModeColorFilterCompat(int,androidx.core.graphics.BlendModeCompat)). It will automatically fallback to PorterDuff for OS versions that don't support BlendMode.

Reply:
I will also add that AntennaPod already uses it [here](https://github.com/AntennaPod/AntennaPod/blob/71880c62d9fc4bbe8553d57cd144d022d204b799/app/src/main/java/de/danoeh/antennapod/fragment/CoverFragment.java#L108).

Reply:
> allows us to use the newer colorfilter method on OS versions that support it

What does "new" mean there? What is the advantage of using `BlendModeColorFilter`? Both seem to have the same visual effect and `PorterDuffColorFilter` needs less code with fewer indirections.

Reply:
Sorry for lateness.

Honestly I do agree. This is where Android documentation really needs to step up its game. It really doesn't explain anywhere why they suggest using `BlendModeColorFIlter` on API 29 and up, or what the difference even is between `BlendModeColorFIlter` and `PorterDuffColorFilter`. Perhaps they plan on deprecating `PorterDuffColorFilter` in the future, who knows.

Reply:
> This is where Android documentation really needs to step up its game. It really doesn't explain anywhere why they suggest using BlendModeColorFIlter on API 29 and up, or what the difference even is between BlendModeColorFIlter and PorterDuffColorFilter. 

Yeah, I tried to find the difference as well and did not find any documentation.

> Perhaps they plan on deprecating PorterDuffColorFilter in the future, who knows.

I would then switch if/when they actually deprecate it. Until then, we should use the direct method without compatibility layer.